zstacklib/zstacklib/utils/upload_task.py

```
@@ -0,0 +1,409 @@
+        up.slice_size = get_long_field('X-SLICE-SIZE', default=up.image_size)
+        up.slice_index = get_long_field('X-SLICE-INDEX', default=0)
+        up.expected_md5 = req_header.get('X-SLICE-MD5', None)
+
+        if up.slice_offset >= up.image_size:
+            raise Exception('invalid slice offset header: %s, image_size: %d' % (up.slice_offset, up.image_size))
+
+        return up
+
+    def get_upload_task(self, param):
+        # type: (UploadParam) -> UploadTask
+        task = upload_tasks.get_task(param.image_uuid)
```

Ruff (0.13.1)

Undefined name `upload_tasks`

(F821)